### ISSUE TITLE
isXssFound() returns false if xss string was found in array value that is not the last one.

### ISSUE THREAD

**Issue by dlt-** [voku/anti-xss#75]
```
        $test = ['key1'=>"<script>alert('hax');</script>"];
        $antiXss->xss_clean($test);
        var_dump($antiXss->isXssFound());

        $test = ['key1'=>"<script>alert('hax');</script>", 'key2'=>"test"];
        $antiXss->xss_clean($test);
        var_dump($antiXss->isXssFound());
                
        $test = ["key1"=>"test", 'key2'=>"<script>alert('hax');</script>"];
        $antiXss->xss_clean($test);
        var_dump($antiXss->isXssFound());
```

This should return true for each one. But incorrecly reports second test as false:
```
bool(true)
bool(false)
bool(true)
```

This is likely a bug in recursive array check which incorrectly resets _xss_found boolean in each iteration.

**Comment by Anyqax** [comment]
We just stumbled over this as well, seems like a rather critical bug to me if anyone relies on the isXssFound() return value before cleaning as we did.

**Comment by voku** [comment]
Thanks for the error report, fixed in version 4.1.33

**Comment by voku** [comment]
Referred to the wrong issue, sorry.
